### PR TITLE
add proguard rules to fix jackson deserialization

### PR DESCRIPTION
**Pull request by ajwootto**

### CODE REVIEW

**Review by kaushalkapasi — APPROVED** [review]

File: android-client-sdk/proguard-rules.pro [hunk 1]
```diff
@@ -18,4 +18,6 @@
 
 # If you keep the line number information, uncomment this to
 # hide the original source file name.
-#-renamesourcefileattribute SourceFile
\ No newline at end of file
+#-renamesourcefileattribute SourceFile
+-keep class kotlin.Metadata { *; }
```
Comment (VicV) [inline android-client-sdk/proguard-rules.pro:22]: /is this necessary

File: android-client-sdk/proguard-rules.pro [hunk 2]
```diff
@@ -18,4 +18,6 @@
...
+-keep class com.devcycle.sdk.android.model.** { *; }
```
Comment (VicV) [inline android-client-sdk/proguard-rules.pro:23]: also lets make this part of the consumer proguard files so it ships with the sdk now

Comment (ajwootto) [reply android-client-sdk/proguard-rules.pro:23]: oh yeah i forgot that

File: android-client-sdk/proguard-rules.pro (hunk 1 above)
Comment (ajwootto) [reply android-client-sdk/proguard-rules.pro:22]: Im not sure but its recommended in the jackson docs

**Pull request merged**